mobile/ios/ios-device.ts

```diff
@@ -320,11 +327,130 @@ export class IOSDevice implements Mobile.IIOSDevice {
 			installationProxy.deployApplication(packageFile).wait();
 			installationProxy.closeSocket();
 		}).future<void>()();
+    }
+
+    private openSafariFrontEnd(packageName: string): void {
+        var tnsIosPackage = this.$npm.install("tns-ios").wait();
+        var safariPath = path.join(tnsIosPackage, "WebInspectorUI/Safari/Main.html");
+        this.$childProcess.exec("open -a Safari " + safariPath).wait();
+    }
+
+    private attachWhenReady(packageName: string,
+        notificationProxyClient: iOSProxyServices.NotificationProxyClient): void {
```

Comment:
Only the debug part requires the notificationproxyClient. The whole debugging logic must be refactored and extracted in a separate class where the notificationProxy will be a field. It will increase the cohesion. I don't like adding a field for something that is used by a quarter of the class methods.